ncclient/transport/ssh.py

```diff
@@ -381,12 +395,32 @@ def connect(self, host, port=830, timeout=None, unknown_host_cb=default_unknown_
         except paramiko.SSHException:
             raise SSHError('Negotiation failed')
 
-        # host key verification
-        server_key = t.get_remote_server_key()
-
-        fingerprint = _colonify(hexlify(server_key.get_fingerprint()))
+        # Nonstandard port - append to hostname for host pubkey checking
+        if port is not 22:
+            host = '[%s]:%d' % (host, port)
 
-        if hostkey_verify:
+        if hostkey:
+            key_object = None
+            # Since each key (base64-decoded) begins with the key type e.g. 'ssh-ed25519',
+            # we don't need to specify/pass it in; instead leave for the paramiko key class to check,
+            # which raises SSHException if the key type within the key doesn't match that for the class
+            for key_cls in [paramiko.RSAKey, paramiko.DSSKey, paramiko.ECDSAKey, paramiko.Ed25519Key]:
+                try:
+                    key_object = key_cls(data=base64.b64decode(hostkey))
+                except paramiko.SSHException as e:
+                    # Not a key of this type - try the next
+                    pass
+            if not key_object:
+                # We've tried all known host key types and haven't found which one to use - bail
+                raise e
+            # Add the supplied hostkey to my host_keys, to be used in the following .check()
+            # key_object.get_name() is actually the key type e.g. 'ssh-rsa'
+            self._host_keys.add(host, key_object.get_name(), key_object)
```

Comment:
One of the checkin comments says "Add ability to verify remote host against specific hostkey", but the way the code is written means that the provided hostkey will be _added_ to any existing hostkeys? Is that the intent?

Reply:
Correct - will change thanks